1/backend/src/resolvers.js

```diff
@@ -1,67 +1,268 @@
-                if (tod){
-                    tod.message = args.message;
-                    return tod;
+        editTodo: async (parent, args, context) => {
+            if (args.id != null && args.message != null && context.token) {
+                const {driver} = context
+                const editTodoCypher = `
+                    MATCH (todo:Todo {id: $id})
+                    SET todo.message = $message
+                    RETURN todo.id, todo.message, todo.completed
+                `
+                const session = driver.session()
+                try {
+                    const _response = await session.run(editTodoCypher, {id: args.id, message: args.message})
+                    const [response] = await _response.records.map(record => ({
+                        id: record.get('todo.id'),
+                        message: record.get('todo.message'),
+                        completed: record.get('todo.completed')
+                    }))
+                    if (response) {
+                        return response
+
+                    }
+                    return null
+                } finally {
+                    await session.close()
                 }
-                return
             }
 
         },
-        finishTodo: (parent, args, context) => {
-            if (args.id != null && context.token){
-                let tod = data.todos.find(td => td.id === parseInt(args.id));
-                if (tod){
-                    //Reverse status of item
-                    tod.completed = !tod.completed;
-                    return tod
+        finishWithMerge: async (parent, args, context) => {
+            if (args.id != null && context.token) {
+                const {driver} = context
+                const mergeTodoCypher = `
+                    MERGE (todo:Todo {id: $id})
+                    ON MATCH SET todo.completed = True
+                    RETURN todo.id, todo.message, todo.completed
+                `
+                const session = driver.session()
+                try {
+                    const _response = await session.run(mergeTodoCypher, {id: args.id})
+                    const [response] = await _response.records.map(record => ({
+                        id: record.get('todo.id'),
+                        message: record.get('todo.message'),
+                        completed: record.get('todo.completed')
+                    }))
+                    if (response) {
+                        return response
+
+                    }
+                    return null
+                } finally {
+                    await session.close()
+                }
+            }
+
+        },
+        assignTodoToUser: async (parent, args, context) => {
+            if (args.id != null && args.user != null && context.token) {
+                const {driver} = context
+                const assignCypher = `
+                    MATCH (u:User), (t:Todo)
```

Comment:
Have you checked the performance of this on http://localhost:7474/? I think this would create a cartesian product which is undesirable.